### PR TITLE
#158 Allowlist of Ruuter services exposed as public REST endpoints

### PR DESCRIPTION
**Pull request by RayDNoper**
Added functionality to limit access to specific internal DSL's to specific IPs and referrers.

### CODE REVIEW

File: samples/CONFIGURATION.md [hunk 1]
```diff
@@ -155,3 +155,17 @@ defaultServiceInCaseOfException:
 
 There is an example for POST requests. To add default headers to other requests in application.yml, define it in application.yml, also in
 ApplicationProperties class the same way as HttpPost is defined and use addHeaders method from HttpQueryArgs to add these default headers to the request.
+
+
+### Internal requests
+
+Some DSL services can be exposed to to internal requests as REST endpoints.
```
Comment (turnerrainer) [inline samples/CONFIGURATION.md:162]:
```suggestion
Some DSL services can be exposed to internal requests as REST endpoints.
```

File: samples/CONFIGURATION.md [hunk 2]
```diff
@@ -155,3 +155,17 @@ defaultServiceInCaseOfException:
...
+These can be accessed only from IPs that are specified in `allowedFiletypes` configuration block:
+* `allowedIPs` - list of whitelisted IPs 
+* `allowedURLs` - list of whitelisted referral URLs
+
+```
+  internalRequests:
+    allowedIPs: ["127.0.0.1", "192.168.0.1", "172.21.0.1"]
+    allowedURLs: ["http://localhost/internalTest"]
```
Comment (turnerrainer) [inline samples/CONFIGURATION.md:170]: Do these work when using ports as well? For example, `127.0.0.1:8081` and `http://localhost:8082/internalTest`.

File: samples/CONFIGURATION.md [hunk 3]
```diff
@@ -155,3 +155,17 @@ defaultServiceInCaseOfException:
 
 There is an example for POST requests. To add default headers to other requests in application.yml, define it in application.yml, also in
 ApplicationProperties class the same way as HttpPost is defined and use addHeaders method from HttpQueryArgs to add these default headers to the request.
+
+
+### Internal requests
+
+Some DSL services can be exposed to to internal requests as REST endpoints.
+These can be accessed only from IPs that are specified in `allowedFiletypes` configuration block:
```
Comment (turnerrainer) [inline samples/CONFIGURATION.md:163]: Specified in `allowedFiletypes` or `internalRequests`?

**Review by turnerrainer — CHANGES_REQUESTED** [review]
Some comments need addressing before merging.

File: samples/CONFIGURATION.md (hunk 2 above)
Comment (RayDNoper) [reply samples/CONFIGURATION.md:170]: For IPs, the port can not be specified (as outgoing request port can vary). For referrer URL, the port can be specified.

File: samples/CONFIGURATION.md (hunk 3 above)
Comment (RayDNoper) [reply samples/CONFIGURATION.md]: Fixed typo.

**Comment by RayDNoper** [comment]
As I cannot review my own PR, merged without formal approval after addressing the  questions.

**Pull request merged**